gptel-bedrock.el

```diff
@@ -513,13 +513,34 @@ Non-nil CLEAR-CACHE will refresh credentials."
        (gptel-bedrock--fetch-aws-profile-credentials profile t))
       (t (user-error "AWS credentials expired for profile: %s" profile)))))
 
+(defun gptel-bedrock--fetch-instance-credentials ()
+  "Fetch AWS credentials using aws configure export-credentials.
+
+Returns a list of access key, secret key, and session token."
+  (with-temp-buffer
+    (condition-case nil
+        (progn
+          (call-process "aws" nil t nil "configure" "export-credentials")
+          (unless (zerop (buffer-size))
+            (let* ((creds (json-parse-string (buffer-string) :object-type 'plist))
```

Comment:
1. Prefer buffer parsing to buffer->string->json-parse.
2. Use `gptel--json-read`, which is just `json-parse-buffer` with `:object-type 'plist`.